compiler/src/yul/mappers/module.rs

```diff
@@ -0,0 +1,55 @@
+use crate::errors::CompileError;
+use crate::yul::mappers::contracts;
+use crate::yul::namespace::scopes::{ModuleScope, Shared};
+use crate::yul::namespace::types;
+use std::rc::Rc;
+use vyper_parser::ast as vyp;
+use yultsur::yul;
+
+/// Builds a vector of Yul contracts from a Vyper module.
+pub fn module(module: &vyp::Module) -> Result<Vec<yul::Object>, CompileError> {
+    let scope = ModuleScope::new();
+
+    Ok(module
+        .body
+        .iter()
+        .map(|stmt| module_stmt(Rc::clone(&scope), &stmt.node))
+        .collect::<Result<Vec<Option<yul::Statement>>, CompileError>>()?
+        .into_iter()
+        .filter_map(|statement| {
+            if let Some(yul::Statement::Object(object)) = statement {
+                return Some(object);
+            }
+
+            None
+        })
+        .collect::<Vec<yul::Object>>())
```

Comment:
consider if we can simplify this by having `module_stmt` returning an `Option<yul::Statement>` as in other places

Reply:
Maybe... `Vec<yul::Object>` is wrong and should at least be replaced with `HashMap<String, yul::Object>`, `yul::Statement`, or `yul::Object`. At the moment, I think `yul::Statement` is best. In the case of a contract-less module, we would just return the Yul statement `object "MyModule" { }` and with multiple contracts, we would return:

```
object "MyModule" {
  object "MyContract1" { ... }
  object "MyContract2" { ... }
}
```

I'll need to dig a bit deeper into Yul to figure out what's most appropriate. With Solc, I'm pretty sure it's possible to get the bytecode for nested Yul objects after compiling.

Reply:
I'll need to look into this further. Being able to return a single statement depends on whether or not we can retrieve the bytecode for each individual contract contained within that statement after compiling...

If this is not possible, it's probably better to return `HashMap<String, yul::Object>`, where `String` is a contract name.